### ISSUE TITLE
The app doesn't support Arabic subtitle

### ISSUE THREAD

**Issue by HaremHasan** [iina/iina#3328]
**System and IINA version:**

- macOS 
- IINA 

**Expected behavior:**

**Actual behavior:**

<details>
<summary><b>Crash report:</b></summary>

```
```
</details>

<br>

<details>
<summary><b>mpv log:</b></summary>

```
```
</details>

<br>

**Steps to reproduce:**

- [ ] MPV does not have this problem.

**How often does this happen?**
Every time

**Comment by Fujiwara-Chika** [comment]
[libass.9.dylib.zip](https://github.com/iina/iina/files/6009302/libass.9.dylib.zip)
Try this patch. It can solve most subtitle problems.
Unzip and move it to `/Applications/IINA.app/Contents/Frameworks`
<img width="1188" alt="iShot2021-02-19 18 31 32" src="https://user-images.githubusercontent.com/44722787/108493427-44a7fc80-72e1-11eb-884c-c9cdb90ddf9d.png">

**Comment by HaremHasan** [comment]
> [libass.9.dylib.zip](https://github.com/iina/iina/files/6009302/libass.9.dylib.zip)
> 
> Try this patch. It can solve most subtitle problems.
> 
> Unzip and move it to `/Applications/IINA.app/Contents/Frameworks`
> 
> <img width="1188" alt="iShot2021-02-19 18 31 32" src="https://user-images.githubusercontent.com/44722787/108493427-44a7fc80-72e1-11eb-884c-c9cdb90ddf9d.png">
> 
> 
> 
> 
> 
> 

Thank you for your reply. But I do this and still dosn't work with me!!!!![image](https://user-images.githubusercontent.com/44499901/108498477-fa615400-72bd-11eb-9fdd-2f0efee65ae2.jpeg)

**Comment by Fujiwara-Chika** [comment]
Did you install IINA by `home-brew`?

**Comment by HaremHasan** [comment]
> Did you install IINA by `home-brew`?🤔

No. I downloaded the app from website of the app. Then I'm installed dmg file as a regular

**Comment by Fujiwara-Chika** [comment]
> > Did you install IINA by `home-brew`?🤔
> 
> No. I downloaded the app from website of the app. Then I'm installed dmg file as a regular

It's strange🤪. My friend ( at Big Sur) and I ( at Mojave ) both can solve the issue by the patch file.

**Comment by HaremHasan** [comment]
> > > Did you install IINA by `home-brew`?🤔
> 
> > 
> 
> > No. I downloaded the app from website of the app. Then I'm installed dmg file as a regular
> 
> 
> 
> It's weird🤪. My friend ( at Big Sur) and I ( at Mojave ) both solved the issue by the patch file.
> 
> Are you using a M1 Mac?

Yes I have Macbook Air M1 !!

**Comment by Fujiwara-Chika** [comment]
The patch is for Intel Mac, maybe you should compile libass `-with-fontconfig --disable-coretext` for AMR Mac alone, but I'm not a dev, don't know how to do it😅

**Comment by Yamatotsu** [comment]
Hello @Fujiwara-Chika , thank you for the answers ! I tried the patch you gave above but it crashes the app when I replace the libass file. Any ideas ? (Mac OS 11.2 beta, I5 2018, IINA v.1.1.2)

**Comment by Fujiwara-Chika** [comment]
> Hello @Fujiwara-Chika , thank you for the answers ! I tried the patch you gave above but it crashes the app when I replace the libass file. Any ideas ? (Mac OS 11.2 beta, I5 2018, IINA v.1.1.2)

Try this 
[libass.9.dylib.zip](https://github.com/iina/iina/files/6105426/libass.9.dylib.zip)

**Comment by Fujiwara-Chika** [comment]
> > > Hello @Fujiwara-Chika , thank you for the answers ! I tried the patch you gave above but it crashes the app when I replace the libass file. Any ideas ? (Mac OS 11.2 beta, I5 2018, IINA v.1.1.2)
> > 
> > 
> > Try this
> > [libass.9.dylib.zip](https://github.com/iina/iina/files/6105426/libass.9.dylib.zip)
> 
> Thanks a lot ! The app doesn't crash but subtitles does not appear (any language subs), do you have an idea ? I changed the encoding but that didn't helped... Any other solution ?

In fact, this is the old version of the patch that my friend sent me before. I haven't updated the macOS and IINA for a long time. I'm not a dev, I don't know how to compile a new version.😅

**Comment by Mikasa-san** [comment]
Use IINA version 1.1.0

**Comment by TheSpyGorillas** [comment]
i have IINA version 1.1.2, how can i make arabic subtitles work?

**Comment by godfatherrx** [comment]
> > > > > > Hello @Fujiwara-Chika , thank you for the answers ! I tried the patch you gave above but it crashes the app when I replace the libass file. Any ideas ? (Mac OS 11.2 beta, I5 2018, IINA v.1.1.2)
> > > > > 
> > > > > 
> > > > > Try this
> > > > > [libass.9.dylib.zip](https://github.com/iina/iina/files/6105426/libass.9.dylib.zip)
> > > > 
> > > > 
> > > > Thanks a lot ! The app doesn't crash but subtitles does not appear (any language subs), do you have an idea ? I changed the encoding but that didn't helped... Any other solution ?
> > > 
> > > 
> > > In fact, this is the old version of the patch that my friend sent me before. I haven't updated the macOS and IINA for a long time. I'm not a dev, I don't know how to compile a new version.😅
> > 
> > 
> > Thank you anyway :) Just out of curiosity and because I really need it, which version of IINA do you have ? I'll try again with the same as yours and see if this works !
> 
> IINA 0.015 Build 68 on Mojave.
> 
> > It's weird🤪. My friend ( at Big Sur) and I ( at Mojave ) both solved the issue by the first patch file with IINA v1.1.2.
> 
> Just a hint, maybe you should install `fontconfig` firstly, I don't know ... but I have it
> <img alt="Xnip2021-03-09_22-26-19" width="714" src="https://user-images.githubusercontent.com/44722787/110485425-94325900-8126-11eb-92bd-6923976b2905.png">

yah it solve when I add font you can see the solution for that in this image below , you just need to add adobe Arabic font and encoding then will work perfect 
<img width="1440" alt="Screen Shot 2021-04-24 at 3 51 28 AM" src="https://user-images.githubusercontent.com/46424423/115941995-80616d00-a4b0-11eb-823d-543ec0719510.png">

**Comment by mgabouarab** [comment]
just change the font for any font for arabic ... like Cairo font
